e2e-tests/functions

```diff
@@ -1,15 +1,117 @@
 #!/bin/bash
 
 exec 5>&2
-BASH_XTRACEFD="5"
-
-GIT_COMMIT=$(git rev-parse HEAD)
-BUILD_TIME=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
-GIT_BRANCH=${VERSION:-$(git rev-parse --abbrev-ref HEAD | sed -e 's^/^-^g; s^[.]^-^g;' | tr '[:upper:]' '[:lower:]')}
-IMAGE=${IMAGE:-"perconalab/percona-server-mysql-operator:${GIT_BRANCH}"}
-IMAGE_MYSQL=${IMAGE_PXC:-"perconalab/percona-server-mysql-operator:main-pxc8.0"}
-IMAGE_PMM=${IMAGE_PMM:-"perconalab/pmm-client:dev-latest"}
-SKIP_BACKUPS_TO_AWS_GCP=${SKIP_BACKUPS_TO_AWS_GCP:-1}
-tmp_dir=$(mktemp -d)
-sed=$(which gsed || which sed)
-date=$(which gdate || which date)
+export BASH_XTRACEFD="5"
+
+# set root repo relatively to a test dir
+ROOT_REPO=${ROOT_REPO:-$(realpath ../../..)}
+source "${ROOT_REPO}/e2e-tests/vars.sh"
+
+deploy_operator() {
+	kubectl -n "${NAMESPACE}" apply -f "${DEPLOY_DIR}/crd.yaml"
+	kubectl -n "${NAMESPACE}" apply -f "${DEPLOY_DIR}/rbac.yaml"
+
+	yq eval \
+		"$(printf 'select(documentIndex==1).spec.template.spec.containers[0].image="%s"' "${IMAGE}")" \
+		"${DEPLOY_DIR}/operator.yaml" \
+		| kubectl -n "${NAMESPACE}" apply -f -
+
+	kubectl -n "${NAMESPACE}" apply -f "${TESTS_CONFIG_DIR}/secrets.yaml"
+	kubectl -n "${NAMESPACE}" apply -f "${TESTS_CONFIG_DIR}/client.yaml"
+}
+
+get_cr() {
+	local test_name=$(basename "$(pwd)")
+	yq eval "$(printf '.metadata.name="%s"' "${test_name}")" "${DEPLOY_DIR}/cr.yaml" \
+		| yq eval "$(printf '.spec.crVersion="%s"' "${CR_VERSION}")" - \
+		| yq eval "$(printf '.spec.mysql.image="%s"' "${IMAGE_MYSQL}")" - \
+		| yq eval "$(printf '.spec.orchestrator.image="%s"' "${IMAGE_ORCHESTRATOR}")" -
+}
+
+run_mysql() {
+	local command="$1"
+	local uri="$2"
+
+	kubectl -n "${NAMESPACE}" exec mysql-client -- \
+		bash -c "printf '%s\n' \"${command}\" | mysql -sN $uri" 2>&1 \
+		| sed -e 's/mysql: //' \
+		| (grep -v 'Using a password on the command line interface can be insecure.' || :)
+}
+
+run_curl() {
+	kubectl -n "${NAMESPACE}" exec mysql-client -- bash -c "curl -s -k $*"
```

Comment:
`$@` in most cases pass multiple arguments (with spaces) more correctly than `$*` 
```suggestion
	kubectl -n "${NAMESPACE}" exec mysql-client -- bash -c "curl -s -k $@"
```